samples/snippets/src/main/java/com/example/spanner/CreateDatabaseWithEncryptionKey.java

```diff
@@ -0,0 +1,101 @@
+import com.google.cloud.spanner.Database;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.encryption.EncryptionConfigs;
+import com.google.spanner.admin.database.v1.CreateDatabaseMetadata;
+import java.util.Arrays;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateDatabaseWithEncryptionKey {
+
+  static void createDatabaseWithEncryptionKey() {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    String databaseId = "my-database";
+    String kmsKeyName = "my-key";
+
+    try (Spanner spanner =
+        SpannerOptions.newBuilder().setProjectId(projectId).build().getService()) {
+      DatabaseAdminClient adminClient = spanner.getDatabaseAdminClient();
+      createDatabaseWithEncryptionKey(
+          adminClient,
+          projectId,
+          instanceId,
+          databaseId,
+          kmsKeyName);
+    }
+  }
+
+  static Void createDatabaseWithEncryptionKey(DatabaseAdminClient adminClient,
+      String projectId, String instanceId, String databaseId, String kmsKeyName) {
+    final Database databaseToCreate = adminClient
+        .newDatabaseBuilder(DatabaseId.of(projectId, instanceId, databaseId))
+        .setEncryptionConfig(EncryptionConfigs.customerManagedEncryption(kmsKeyName))
+        .build();
+    final OperationFuture<Database, CreateDatabaseMetadata> operation = adminClient
+        .createDatabase(databaseToCreate, Arrays.asList(
+            "CREATE TABLE Singers ("
+                + "  SingerId   INT64 NOT NULL,"
+                + "  FirstName  STRING(1024),"
+                + "  LastName   STRING(1024),"
+                + "  SingerInfo BYTES(MAX)"
+                + ") PRIMARY KEY (SingerId)",
+            "CREATE TABLE Albums ("
+                + "  SingerId     INT64 NOT NULL,"
+                + "  AlbumId      INT64 NOT NULL,"
+                + "  AlbumTitle   STRING(MAX)"
+                + ") PRIMARY KEY (SingerId, AlbumId),"
+                + "  INTERLEAVE IN PARENT Singers ON DELETE CASCADE"
+        ));
+    try {
+      System.out.println("Waiting for operation to complete...");
+      Database createdDatabase = operation.get(120, TimeUnit.SECONDS);
+
+      System.out.printf(
+          "Database %s created with encryption key %s%n",
+          createdDatabase.getId(),
+          createdDatabase.getEncryptionConfig().getKmsKeyName()
+      );
+    } catch (ExecutionException e) {
+      // If the operation failed during execution, expose the cause.
+      throw (SpannerException) e.getCause();
```

Comment:
It could in theory be a different exception than a `SpannerException`, so this would be safer:

```suggestion
      throw SpannerExceptionFactory.asSpannerException(e.getCause());
```